src/protobuf/java/com/datadoghq/sketch/ddsketch/mapping/IndexMappingProtoBinding.java

```diff
@@ -0,0 +1,63 @@
+package com.datadoghq.sketch.ddsketch.mapping;
+
+import com.datadoghq.sketch.ddsketch.proto.IndexMapping;
+
+public class IndexMappingProtoBinding {
+
+    public static IndexMapping toProto(com.datadoghq.sketch.ddsketch.mapping.IndexMapping indexMapping) {
+        if (indexMapping instanceof LogLikeIndexMapping) {
+            return toProtoLogLike((LogLikeIndexMapping) indexMapping);
+        } else if (indexMapping instanceof BitwiseLinearlyInterpolatedMapping) {
+            return toProtoBitwiseLinear((BitwiseLinearlyInterpolatedMapping) indexMapping);
+        } else {
+            throw new IllegalArgumentException("Unknown indexmapping " + indexMapping.getClass());
+        }
+    }
+
+    private static IndexMapping toProtoLogLike(LogLikeIndexMapping indexMapping) {
+        return com.datadoghq.sketch.ddsketch.proto.IndexMapping.newBuilder()
+                .setGamma(indexMapping.gamma())
+                .setIndexOffset(indexMapping.indexOffset())
+                .setInterpolation(interpolation(indexMapping.interpolation()))
+                .build();
+    }
+
+    private static IndexMapping toProtoBitwiseLinear(BitwiseLinearlyInterpolatedMapping indexMapping) {
+        return com.datadoghq.sketch.ddsketch.proto.IndexMapping.newBuilder()
+                .setGamma(indexMapping.gamma())
+                .setInterpolation(com.datadoghq.sketch.ddsketch.proto.IndexMapping.Interpolation.LINEAR)
+                .build();
+    }
+
+    static public LogLikeIndexMapping fromProto(com.datadoghq.sketch.ddsketch.proto.IndexMapping proto) {
```

Comment:
Should we make it public?

Reply:
This is public, but in non canonical order